src/Serialization/SerializationWriterToStringTrait.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Microsoft\Kiota\Abstractions\Serialization;
+
+use DateInterval;
+use DateTime;
+use DateTimeInterface;
+
+/**
+ * This Trait contains common to-string conversion for common types.
+ */
+trait SerializationWriterToStringTrait
+{
+    public function getDateIntervalValueAsString(DateInterval $value): string
+    {
+        $year = $value->y > 0 ? "%yY" : "";
+        $month = $value->m > 0 ? "%mM" : "";
+        $day = $value->d > 0 ? '%dD' : "";
+        $hour = $value->h > 0 ? '%hH' : "";
+        $minute = $value->i > 0 ? '%iM' : "";
+        $second = $value->s > 0 ? '%sS' : "";
+        $timePart = $hour.$minute.$second;
+        $time = !empty($timePart) ? "T$timePart" : '';
+        return $value->format("%rP$year$month{$day}$time");
+    }
+    public function getDateTimeValueAsString(DateTime $value): string
+    {
+        return $value->format(DateTimeInterface::RFC3339);
+    }
+
+    public function getBooleanValueAsString(bool $value): string
+    {
+        return $value ? 'true' : 'false';
+    }
+
+    public function getEscapedString(string $value): string
```

Comment:
Could be  better names as `getStringValueAsEscapedString` to be consistent with the other methods in the class